### PR TITLE
Analytics Event: `LOAD_MORE_RESULTS`

### LINKED ISSUE #1074: Analytics event: `LOAD_MORE_RESULTS`
## Description

The goal is to 

- add the type for event `LOAD_MORE_RESULTS` to the `Events` type.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/types/analytics.ts#L12

- fire this event from the appropriate location in the codebase using the `useAnalytics` composable.
  https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/composables/use-analytics.ts#L11

## Additional information

- [RFC](https://github.com/WordPress/openverse/blob/main/rfcs/analytics/20221006-implementation_plan_frontend.md#events)
- [Example usage in `VHomeGallery.vue`](https://github.com/WordPress/openverse/blob/839418fabb0972ec3c3ccf48c72fd64cbba03cb9/frontend/src/components/VHomeGallery/VHomeGallery.vue#L128-L134)

### LINKED ISSUE #1992: Add a Playwright fixture that allows retrieving POST request parameters
## Problem

Currently, we use the following code to extract the Analytics payloads from POST requests in Playwright tests:
```
 page.on("request", (req) => {
    if (req.method() === "POST") {
      const requestData = req.postDataJSON()
      if (requestData?.n == "SELECT_EXTERNAL_SOURCE") {
        eventData = JSON.parse(requestData?.p)
      }
    }
  })
```
This code is repeated in every test that tests analytics.

## Description

Abstract this out into a utility or [custom Playwright fixture](https://playwright.dev/docs/test-fixtures#creating-a-fixture) that provides this more generally (like a list of all events that get sent that can be asserted against).

## Additional context

Based on [quote from @sarayourfriend 's comment](https://github.com/WordPress/openverse/pull/1158#discussion_r1183136611)

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #1074 by @dhruvkb 
Fixes #1992 by @sarayourfriend 

## Description

This PR adds a `LOAD_MORE_RESULTS` analytics event to the load more button. ~Crucially, I did _not_ include the original page number property in the event payload, as it requires major additions to the media and search stores.~ Thanks, @obulat.

## Testing Instructions

CI should pass, and then review the code. Let me know what you think of this API surface! Should we even be testing analytics events in e2e tests? I suspect it could be a nice additional assurance. If so we might want to add it in other places, like for pageviews and other custom events. That gets into larger issues about our frontend test consistency and coverage though.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [x] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
I just remembered @obulat added some Playwright analytics testing previously:

https://github.com/WordPress/openverse/blob/9471728b4686d054493cd4528554e957dfee4350/frontend/test/playwright/e2e/external-sources.spec.ts

I'll try updating that implementation to use this new code, which would be a good way to validate its usefulness!

File: frontend/test/playwright/e2e/load-more.spec.ts [hunk 1]
```diff
@@ -9,6 +9,11 @@ import { mockProviderApis } from "~~/test/playwright/utils/route"
 
 import { AUDIO, IMAGE, SupportedMediaType } from "~/constants/media"
 
+import {
+  collectAnalyticsEvents,
+  expectEventPayloadToMatch,
+} from "../utils/analytics"
```
Comment (obulat) [inline frontend/test/playwright/e2e/load-more.spec.ts:15]:
```suggestion
} from "~/utils/analytics"
```

File: frontend/test/playwright/e2e/load-more.spec.ts [hunk 2]
```diff
@@ -12,7 +12,7 @@ import { AUDIO, IMAGE, SupportedMediaType } from "~/constants/media"
 import {
   collectAnalyticsEvents,
   expectEventPayloadToMatch,
-} from "../utils/analytics"
+} from "~/utils/analytics"
```
Comment (obulat) [inline frontend/test/playwright/e2e/load-more.spec.ts]:
```suggestion
} from "~~/test/playwright/utils/analytics"
```
Sorry, I didn't see that this was in tests

**Comment by sarayourfriend** [comment]
> how many times in a session a user + query + mediaType

Can you clarify how this would work? Plausible only allows querying by a single custom prop at a time ([see the note here](https://plausible.io/docs/stats-api#custom-props)).

**Comment by obulat** [comment]
> > how many times in a session a user + query + mediaType
> 
> Can you clarify how this would work? Plausible only allows querying by a single custom prop at a time ([see the note here](https://plausible.io/docs/stats-api#custom-props)). We could work around that by combining the media type and query into a single prop. The media type is critical information for all queries anyway.

I've added the `currentPage` to the media store in `REACH_RESULT_END` event PR, and the plan is to add that in here, too. So we won't need the workaround anymore.

**Comment by sarayourfriend** [comment]
@zackkrida According to @obulat's comment, I suppose this needs to be rebased now that #2454 is merged before it can be reviewed?

**Comment by zackkrida** [comment]
Drafting to implement the `currentPage` functionality merged in #2454

File: frontend/test/playwright/e2e/load-more.spec.ts [hunk 3]
```diff
@@ -158,4 +158,34 @@ test.describe("Load more button", () => {
       })
     })
   }
+
+  /**
+   * Checks that an analytics event is posted to /api/event and has the correct
+   * payload for the LOAD_MORE_RESULTS event.
+   */
+  test(`Sends a valid analytics event when loading more results`, async ({
```
Comment (AetherUnbound) [inline frontend/test/playwright/e2e/load-more.spec.ts:166]: Did we want to have a test for pages beyond 1, or a test where the load more button is visible but not clicked and assert an event is not fired?

**Comment by zackkrida** [comment]
@AetherUnbound and @obulat I've added the currentPage logic and this PR is ready for review.

**Review by AetherUnbound — APPROVED** [review]
This worked as expected! One comment about tests but otherwise this is good to go :)

File: frontend/test/playwright/e2e/load-more.spec.ts (hunk 3 above)
Comment (zackkrida) [reply frontend/test/playwright/e2e/load-more.spec.ts]: ask and ye shall recieve 😄 good idea, added in befd63fcd

**Review by sarayourfriend — APPROVED** [review]
LGTM. Tested just fine locally :+1:

**Pull request merged**